datashape/discovery.py

```diff
@@ -405,3 +405,19 @@ def descendents(d, x):
         children -= desc
         desc.update(children)
     return desc
+
+
+Mock = None
+try:
+    from unittest.mock import Mock
+except ImportError:
+    try:
+        from mock import Mock
+    except ImportError:
+        pass
+
+if Mock is not None:
+    @dispatch(Mock)
```

Comment:
I think we could make an ABC that checks for these two attrtibutes and then use the normal multiple dispatch methods. This could also explicitly guard against mocks there.